server/handlers/attachments_test.go

```diff
@@ -257,3 +259,215 @@ func TestHandleMessageReference(t *testing.T) {
+				client.On("GetFileContent", "").Return([]byte{}, nil)
+			},
+			attachments: []msteams.Attachment{
+				{
+					Name: "mock-name",
+				},
+			},
+			expectedText: "mock-text",
+		},
+		{
+			description: "Error uploading the file",
+			setupPlugin: func(p *mocksPlugin.PluginIface, mockAPI *plugintest.API, client *mocksClient.Client, store *mocksStore.Store) {
+				p.On("GetClientForUser", testutils.GetUserID()).Return(client, nil)
+				p.On("GetAPI").Return(mockAPI)
+			},
+			setupAPI: func(mockAPI *plugintest.API) {
+				mockAPI.On("GetConfig").Return(&model.Config{
+					FileSettings: model.FileSettings{
+						MaxFileSize: model.NewInt64(5),
+					},
+				})
+				mockAPI.On("UploadFile", []byte{}, testutils.GetChannelID(), "mock-name").Return(nil, &model.AppError{Message: "error uploading the file"})
+				mockAPI.On("LogError", "upload file to mattermost failed", "filename", "mock-name", "error", "error uploading the file").Return()
+			},
+			setupClient: func(client *mocksClient.Client) {
+				client.On("GetFileContent", "").Return([]byte{}, nil)
+			},
+			attachments: []msteams.Attachment{
+				{
+					Name: "mock-name",
+				},
+			},
+			expectedText: "mock-text",
+		},
+		{
+			description: "Number of attachments are greater than 10",
+			setupPlugin: func(p *mocksPlugin.PluginIface, mockAPI *plugintest.API, client *mocksClient.Client, store *mocksStore.Store) {
+				p.On("GetClientForUser", testutils.GetUserID()).Return(client, nil)
+				p.On("GetAPI").Return(mockAPI)
+			},
+			setupAPI: func(mockAPI *plugintest.API) {
+				mockAPI.On("GetConfig").Return(&model.Config{
+					FileSettings: model.FileSettings{
+						MaxFileSize: model.NewInt64(5),
+					},
+				})
+				mockAPI.On("UploadFile", []byte{}, testutils.GetChannelID(), "").Return(&model.FileInfo{}, nil)
+				mockAPI.On("LogDebug", "discarding the rest of the attachments as mattermost supports only 10 attachments per post").Return()
+			},
+			setupClient: func(client *mocksClient.Client) {
+				client.On("GetFileContent", "").Return([]byte{}, nil)
+			},
+			attachments: []msteams.Attachment{
+				{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {},
+			},
+			expectedText:               "mock-text",
+			expectedAttachmentIDsCount: 10,
+		},
+		{
+			description: "Attachment type code snippet",
+			setupPlugin: func(p *mocksPlugin.PluginIface, mockAPI *plugintest.API, client *mocksClient.Client, store *mocksStore.Store) {
+				p.On("GetClientForUser", testutils.GetUserID()).Return(client, nil)
+			},
+			setupAPI: func(mockAPI *plugintest.API) {
+				mockAPI.On("GetConfig").Return(&model.Config{
+					FileSettings: model.FileSettings{
+						MaxFileSize: model.NewInt64(5),
+					},
+				})
+				mockAPI.On("UploadFile", []byte{}, testutils.GetChannelID(), "mock-name").Return(&model.FileInfo{}, nil)
+			},
+			setupClient: func(client *mocksClient.Client) {
+				client.On("GetCodeSnippet", "mock-url////////////").Return("", nil)
+			},
+			attachments: []msteams.Attachment{
```

Comment:
There should be a testutil for getting array `Attachment`.

Reply:
not possible because of import cycle